nuclear/cmake/CompilerWarnings.cmake

```diff
@@ -20,30 +20,18 @@ function(set_project_warnings)
       -Wsign-conversion # warn on sign conversions
       -Wdouble-promotion # warn if float is implicit promoted to double
       # -Wformat=2 # warn on security issues around functions that format output (ie printf)
-  )
-
-  if(WARNINGS_AS_ERRORS)
-    set(CLANG_WARNINGS ${CLANG_WARNINGS} -Werror)
-  endif()
-
-  set(GCC_WARNINGS
-      ${CLANG_WARNINGS}
       -Wnull-dereference # warn if a null dereference is detected
       -Wmisleading-indentation # warn if indentation implies blocks where blocks do not exist
-      -Wduplicated-cond # warn if if / else chain has duplicated conditions
-      -Wduplicated-branches # warn if if / else branches have duplicated code
-      -Wlogical-op # warn about logical operations being used where bitwise were probably wanted
-      -Wuseless-cast # warn if you perform a cast to the same type
+      # * We want the warnings below if clang adds support for them
+      # * Until then, these make clang-tidy mad, and we would have to have an extra CI pipeline step to cover them
+      # * -Wlogical-op # warn about logical operations being used where bitwise were probably wanted
+      # * -Wuseless-cast # warn if you perform a cast to the same type
   )
```

Comment:
Clang-tidy already finds these on it's own, so I don't think we need our compilers to also find them so you can just delete these